src/main/kotlin/org/rust/ide/intentions/ExtractInlineModule.kt

```diff
@@ -0,0 +1,63 @@
+package org.rust.ide.intentions
+
+import com.intellij.codeInsight.intention.IntentionAction
+import com.intellij.openapi.command.WriteCommandAction
+import com.intellij.openapi.editor.Editor
+import com.intellij.openapi.project.Project
+import com.intellij.openapi.util.io.FileUtil
+import com.intellij.psi.PsiFile
+import com.intellij.psi.PsiFileFactory
+import com.intellij.psi.impl.file.PsiFileImplUtil
+import com.intellij.psi.util.PsiTreeUtil
+import com.intellij.refactoring.move.moveFilesOrDirectories.MoveFilesOrDirectoriesUtil
+import org.rust.lang.RustLanguage
+import org.rust.lang.core.psi.RustExpr
+import org.rust.lang.core.psi.RustModItem
+import org.rust.lang.core.psi.impl.RustFile
+import org.rust.lang.core.psi.impl.RustModDeclItemImpl
+import org.rust.lang.core.psi.impl.RustModItemImpl
+import org.rust.lang.core.psi.util.RustModules
+import org.rust.lang.core.psi.util.modDecls
+
+class ExtractInlineModule : IntentionAction {
+    override fun getFamilyName() = "Extract inline module structure"
+    override fun getText() = "Extract Inline Module"
+    override fun startInWriteAction() = true
+
+    fun createFileFromText(project: Project, text: String): RustFile? {
+        return PsiFileFactory.getInstance(project).createFileFromText("a.rs", RustLanguage, text) as RustFile?
+    }
+
+    fun createModDeclItem(project: Project, modName: String): RustModItem? {
+        val file = createFileFromText(project, "mod $modName;")
+        return PsiTreeUtil.findChildOfType(file, RustModItem::class.java)
```

Comment:
It might be nice to create more *kotlinish* wrapper for this (using reified type params), similar to [`PsiElement#parentOfType`](https://github.com/intellij-rust/intellij-rust/blob/master/src/main/kotlin/org/rust/lang/core/psi/util/RustPsiExtensions.kt#L18).